### PR TITLE
Move values to enum

### PR DESCRIPTION
**Pull request by brianrbrenner**
This PR moves enum fields of `Values` from structs to enum variants. Also adds additional error handling when creating a new SDL2 platform.

### CODE REVIEW

File: atmos_gen/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,6 @@ edition = "2018"
 proc-macro = true
 
 [dependencies]
-proc-macro2="1.0"
+proc-macro2="1.0.60"
```
Comment (brianrbrenner) [inline atmos_gen/Cargo.toml]: Was failing a build step until explicitly adding this.

**Comment by amshafer** [comment]
Can you clean up the commits to keep a clean history? There should only be two once you've squashed them together: the `Value` change and the SDL error message change. For example the commits to add/remove the useless feature will be dropped. `git rebase` will do all this if you haven't used it before.

File: dakota/src/dom.rs [hunk 2]
```diff
@@ -90,56 +90,43 @@ pub struct Content {
...
+        Ok(match *self {
+            Self::Relative(r) => Self::scale(r, avail_space)? as i32,
+            Self::Constant(c) => c,
         })
     }
+
+    pub fn new(&self) -> Self {
+        match *self {
+            Self::Relative(r) => {
+                assert!(r >= 0.0 && r <= 1.0);
```
Comment (amshafer) [inline dakota/src/dom.rs]: I think we should get rid of the `new` method. It looks pretty weird doing `dom::Value::Constant(resolution.0 as i32).new(),`.

Instead I think we should move this assert into `scale`, and then we can do `dom::Value::Constant(resolution.0 as i32).` and the value check will happen when it's used.

File: dakota/src/lib.rs [hunk 3]
```diff
@@ -307,19 +307,22 @@ impl Dakota {
...
+            match platform::SDL2Plat::new() {
+                Ok(sdl) => {
+                    let mut sdl: Box<dyn Platform> = Box::new(sdl);
+                    match Self::init_thundr(&mut sdl) {
+                        Ok(thundr) => match thundr.get_dpi() {
+                            Ok(dpi) => {
+                                log::error!("Using SDL2 Platform Backend");
+                                return Ok((sdl, thundr, dpi));
+                            }
+                            Err(e) => log::error!("Failed to create SDL2 backend: {:?}", e),
```
Comment (amshafer) [inline dakota/src/lib.rs]: This predates your change but I would change this message to "Failed to get DPI from SDL" to reflect what the error actually takes place during.

File: dakota/src/dom.rs (hunk 2 above)
Comment (brianrbrenner) [reply dakota/src/dom.rs:124]: Is this check not already happening in scale? Only it's a conditional that throws instead off an assertion?

Comment (brianrbrenner) [reply dakota/src/dom.rs]: Isn't this already sort of happening in `scale()` via the conditional that throws if true? Would scale be implicitly called on instantiation?

Comment (amshafer) [reply dakota/src/dom.rs]: Heh I don't even remember  adding that. Then yes that should do it, you can go ahead with removing `new`

Comment (brianrbrenner) [reply dakota/src/dom.rs]: Removing new breaks the `viewports.xml` file. Investigating this now.

File: dakota/src/dom.rs [hunk 4]
```diff
@@ -90,54 +90,31 @@ pub struct Content {
...
-        if !(self.val >= 0.0 && self.val <= 1.0) {
+impl Value {
+    fn scale(current: f32, val: f32) -> Result<u32> {
+        if !(current >= 0.0 && current <= 1.0) {
             return Err(anyhow!(
                 "Element.relativeOffset should use values in the range (0.0, 1.0)"
             ));
         }
-        Ok((val * self.val) as u32)
+        Ok((current * val) as u32)
```
Comment (amshafer) [inline dakota/src/dom.rs]: I think we can also change this to return an i32 to avoid the casting mess. This is only called by `get_value` anyway.

Comment (brianrbrenner) [reply dakota/src/dom.rs]: Updated

**Comment by amshafer** [comment]
I think the squash didn't go smoothly, looks like there's a bunch of extra things mixed up in there?

**Comment by amshafer** [comment]
Thanks!

**Pull request merged**